### PR TITLE
fix(issues-stream): Fix pagination [SEN-138]

### PR DESCRIPTION
**Pull request by billyvg**
This fixes pagination on organization issues stream (sentry 10 only). Cursors were incorrectly being updated when auto updates were on. Also removes cursor when you return to first page so we fetch the latest results.

Fixes SEN-138

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationStream/overview.jsx [hunk 1]
```diff
@@ -317,6 +320,18 @@ const OrganizationStream = createReactClass({
...
+        const pageLinks = jqXHR.getResponseHeader('Link');
+        let initialCursor = this.state.initialCursor;
+
+        // Reset `initialCursor` if current route does not have "cursor" url param
+        // i.e. if you page back to first page and cursor is removed so that we fetch
+        // a fresh set of results, we should reset initial cursor
+        if ((!this.state.initialCursor || !currentQuery.cursor) && pageLinks) {
+          const links = utils.parseLinkHeader(pageLinks);
+          if (!links.previous.results) {
+            initialCursor = links.next.cursor;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: If you are on the 0th page won't this point at the 1st page?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: It does, we then use this cursor to find when we go from page 1 back to page 0, and when we do, we remove the cursor so it fetches an up-to-date set of results.

Comment (markstory) [reply src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: Right. I didn't fully grok the other block the first time around.

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/organizationStream/overview.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: This assumption doesn't quite hold up when you navigate several pages deep. e.g. going from page 0 -> 1 -> 2 -> 1 -> 0, the cursor pointing from 0 -> 1 is not the same from 1 -> 0. I'm going to change this to just track a "page" number throughout.

**Comment by billyvg** [comment]
@markstory I update this to track page numbers in the URL

File: src/sentry/static/sentry/app/views/organizationStream/overview.jsx [hunk 2]
```diff
@@ -415,8 +413,17 @@ const OrganizationStream = createReactClass({
     this.transitionTo({sort});
   },
 
-  onCursorChange(cursor, path, query) {
-    this.transitionTo({cursor});
+  onCursorChange(cursor, path, query, pageDiff) {
+    const queryPageInt = parseInt(query.page, 10);
+    let nextPage = isNaN(queryPageInt) ? pageDiff : queryPageInt + pageDiff;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: In the `-1` case should `nextPage` be 0?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: Oops, fixed so that if nextPage is <= 0, we reset cursor/page.

**Review by markstory — APPROVED** [review]

**Pull request merged**